# DecodeLabs Data Analytics - Project 2

## Exploratory Data Analysis (EDA)

This project focuses on exploring the cleaned dataset to identify patterns, trends, distributions, and outliers.

The analysis will include descriptive statistics, categorical analysis, trend analysis, outlier detection, and key observations from the data.

In [1]:
import pandas as pd
import plotly.express as px

df = pd.read_excel("Cleaned_Dataset_for_Data_Analytics.xlsx")

In [2]:
!pip install plotly


[notice] A new release of pip is available: 25.3 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
!pip install -U nbformat


[notice] A new release of pip is available: 25.3 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [4]:
import plotly.express as px

In [5]:
df.shape


(1200, 14)

In [6]:
df.columns.tolist()

['OrderID',
 'Date',
 'CustomerID',
 'Product',
 'Quantity',
 'UnitPrice',
 'ShippingAddress',
 'PaymentMethod',
 'OrderStatus',
 'TrackingNumber',
 'ItemsInCart',
 'CouponCode',
 'ReferralSource',
 'TotalPrice']

In [7]:
df[["Quantity", "UnitPrice", "ItemsInCart", "TotalPrice"]].describe()

,Quantity,UnitPrice,ItemsInCart,TotalPrice
count,1200.000000,1200.000000,1200.000000,1200.000000
mean,2.945833,356.412750,5.485000,1053.968300
std,1.407557,197.177146,2.281983,819.856558
min,1.000000,11.390000,1.000000,11.390000
25%,2.000000,186.062500,4.000000,410.520000
50%,3.000000,364.210000,5.000000,823.615000
75%,4.000000,521.570000,7.000000,1578.475000
max,5.000000,699.930000,10.000000,3456.400000


## 1. Descriptive Statistics

Descriptive statistics were calculated for the numerical variables `Quantity`, `UnitPrice`, `ItemsInCart`, and `TotalPrice`.

The dataset contains 1,200 observations. The average quantity ordered per transaction is approximately 2.95, with a median of 3 items. The average unit price is 356.41, while the median unit price is 364.21.

Customers have an average of 5.49 items in their cart, with a median of 5 items. The average `TotalPrice` is 1,053.97, while the median is 823.62.

The difference between the mean and median of `TotalPrice` suggests that some transactions have relatively high order values and may require further investigation during the outlier analysis.

In [8]:
df["Product"].value_counts()

Product
Printer    181
Tablet     179
Chair      178
Laptop     173
Desk       170
Monitor    163
Phone      156
Name: count, dtype: int64

## 2. Product Distribution

The distribution of orders across products was analyzed to determine which products appeared most frequently in the dataset.

The Printer was the most frequently ordered product with 181 orders, while Phone had the lowest number of orders with 156. Tablet, Chair, Laptop, Desk, and Monitor recorded 179, 178, 173, 170, and 163 orders respectively.

Overall, the distribution across the seven product categories is relatively balanced, with only a 25-order difference between the most and least frequently ordered products.

In [9]:
df.groupby("Product")["TotalPrice"].sum().sort_values(ascending=False)

Product
Chair      195620.11
Printer    195612.61
Laptop     192126.56
Tablet     186568.95
Monitor    175651.41
Desk       167459.93
Phone      151722.39
Name: TotalPrice, dtype: float64

Product Sales Performance: Chairs generated the highest total sales value at ₦195,620.11, closely followed by Printers at ₦195,612.61. Phones recorded the lowest total sales value at ₦151,722.39. This indicates that Chairs and Printers were the strongest contributors to overall sales revenue, while Phones contributed comparatively less.

In [10]:
product_sales = df.groupby('Product')['TotalPrice'].sum().sort_values(ascending=False)

fig = px.bar(
    x=product_sales.index,
    y=product_sales.values,
    title='Total Sales by Product',
    labels={
        'x': 'Product',
        'y': 'Total Sales'
    }
)

fig.show()

In [11]:
df['Date'] = pd.to_datetime(df['Date'])

In [12]:
monthly_sales = df.groupby(
    df['Date'].dt.to_period('M')
)['TotalPrice'].sum()

monthly_sales

Date
2023-01    56685.75
2023-02    40117.66
2023-03    48609.37
2023-04    27751.71
2023-05    63836.84
2023-06    49500.19
2023-07    42820.66
2023-08    54352.14
2023-09    29526.67
2023-10    52607.85
2023-11    43079.67
2023-12    43754.73
2024-01    38528.08
2024-02    36909.57
2024-03    36030.90
2024-04    49613.14
2024-05    27909.11
2024-06    68068.54
2024-07    42963.98
2024-08    31991.07
2024-09    39794.98
2024-10    37226.97
2024-11    32413.76
2024-12    38785.77
2025-01    29099.40
2025-02    35317.55
2025-03    39200.66
2025-04    31821.20
2025-05    43396.64
2025-06    53047.40
Freq: M, Name: TotalPrice, dtype: float64

### Sales Trend Analysis 
This analysis examines how total sales changed over time to identify noticeable trends, increases, decreases, and fluctuations in sales.

In [13]:
monthly_sales = monthly_sales.reset_index()

monthly_sales['Date'] = monthly_sales['Date'].dt.to_timestamp()

In [14]:
fig = px.line(
    monthly_sales,
    x='Date',
    y='TotalPrice',
    title='Total Sales Over Time',
    labels={
        'Date': 'Date',
        'TotalPrice': 'Total Sales'
    }
)

fig.show()

### Observation

The line graph shows how total sales changed over time. It helps identify periods with higher or lower sales and makes it easier to observe any noticeable trends or fluctuations in sales throughout the dataset.

In [15]:
order_status = df['OrderStatus'].value_counts()

fig = px.bar(
    x=order_status.index,
    y=order_status.values,
    title='Order Status Distribution',
    labels={
        'x': 'Order Status',
        'y': 'Number of Orders'
    }
)

fig.show()

### Observation

The bar chart shows the distribution of orders across the different order statuses. Cancelled and Returned orders have the highest counts, while Delivered orders have the lowest count among the five statuses.

Outlier Analysis

This analysis examines the distribution of total prices to identify unusually high or low order values that may represent potential outliers.

In [16]:
fig = px.box(
    df,
    y='TotalPrice',
    title='Distribution and Outliers of Total Price',
    labels={
        'TotalPrice': 'Total Price'
    }
)

fig.show()

### Observation

The box plot shows the distribution of total prices and helps identify potential outliers. Values appearing beyond the whiskers represent unusually high or low order totals that may require further investigation.